mantle/util/retry.go

```diff
@@ -48,6 +48,27 @@ func RetryConditional(attempts int, delay time.Duration, shouldRetry func(err er
 	return err
 }
 
+func RetryUntilTimeout(timeout, delay time.Duration, f func() error) error {
```

Comment:
Docstring would be useful for this with the exact semantics of `timeout` and `delay`.

Reply:
check the update to see if it's good.